reposcan/exporter.py

```diff
@@ -194,6 +195,62 @@ def dump_errata(self, dump):
                                                 set()).add(repo_id)
                 dump.update(errataid2repoids)
 
+        # Select errata detail for errata API
+        errataid2cves = {}
+        with self._named_cursor() as cursor:
+            cursor.execute("""SELECT errata_cve.errata_id, cve.name
+                                FROM cve
+                                JOIN errata_cve ON cve_id = cve.id
+                           """)
+            for errata_id, cve_name in cursor:
+                errataid2cves.setdefault(errata_id, []).append(cve_name)
+        errataid2pkgs = {}
+        with self._named_cursor() as cursor:
+            cursor.execute("""SELECT pkg_errata.errata_id,
+                                     package_name.name,
+                                     evr.epoch,
+                                     evr.version,
+                                     evr.release,
+                                     arch.name
+                                FROM pkg_errata
+                                JOIN package ON package.id = pkg_errata.pkg_id
+                                JOIN package_name ON package_name.id = package.name_id
+                                JOIN evr ON evr.id = package.evr_id
+                                JOIN arch ON arch.id = package.arch_id
+                           """)
+            for errata_id, p_name, p_epoch, p_ver, p_rel, a_name in cursor:
+                packagename = join_packagename(p_name, p_epoch, p_ver, p_rel, a_name)
```

Comment:
I'd export only errataid to package ids mapping in the dump and then create full package name in webapp using other cached data to make database dump and memory footprint smaller.

Reply:
Altho, the whole point to the export process is to increase the webapp's speed at the expense of memory. In that context, it makes more sense for the export to have as much precomputed as possible.

Reply:
I had the same thought as @ggainey, that pre-computing the package name made sense here for performance reasons.  In other places (updates API) it likely makes sense to keep the package data separate for query filtering/search purposes.

Reply:
Well, it depens if there is any noticeable performance hit caused by building the NEVRA on webapp side.